### DISCUSSION TITLE
How do i make an API call after field validation has passed?

### DISCUSSION THREAD

**Discussion by behrygood1982** [react-hook-form/react-hook-form#9742] [Q&A]
Hello,

We are currently on v6 of react-hook-form.  I have the following requirement.

1. Show an email text box with simple pattern and required validation
2. When an email passes validation, trigger an axios/ajax call to check if an account has already been created for the user.
3. If an account has not been created for the user, show another checkbox input, in the form, below the email text box, asking if they would like to have an account created for them.

Unfortunately, i cannot for the life of me figure out the correct/recommended way of doing this. Another way of putting this would be saying, how do i make a remote call when a certain field/input passes validation and i need to run some extra remote call via ajax/axios.

Thanks

**Comment by bluebill1049** [comment]
could you do that inside `handleSubmit` callback?

**Reply by behrygood1982** [reply]
Hey @bluebill1049 ,

Thanks for responding. I don’t think so. What I want to do it have an email field. When the email field passes required and regex pattern validation, only then I want to make an axios call to the backend to see if the a user already exists with that email. If they don’t, I want to render a checkbox below the input to allow the user to opt for account creation. If the email is already associated to a user the. The checkbox won’t be shown.

**Reply by behrygood1982** [reply]
I want the remote call to check the backend of the user with that email exists only after the validation passes for the email input when the onblur happens

**Reply by bluebill1049** [reply]
oh i see, you can use the `validate` function,

`validate: async () => await fetch('/api')`

**Reply by behrygood1982** [reply]
Okay I tried that but how do I hide the checkbox when the user changes the input to an invalid email? Would u be able to do a code sandbox of this because I cannot figure this out. I spent a couple hours :/

for example, if the user puts a valid email and the remote call returns the user doesn’t exist in the database the checkbox will be shown.  Now if the user changes the email to something that causes regex pattern to fail, how do I now hide the checkbox ( no axios call is made in this scenario )

**Reply by behrygood1982** [reply]
Hey @bluebill1049 ,

I just got home and threw this in a sandbox.  It appears to be working.  Can you let me know if there is a cleaner/better way to achieve this?


https://r6f1qu.csb.app/

I also noticed that when the checkbox is not there `shouldSignUp` is missing from the form data ( since the checkbox is hidden ).  Is there a way to handle make it always show up, that is, exist with value of false when checkbox is not rendered?
